dev_guide/managing_images.adoc

```diff
@@ -1099,6 +1099,62 @@ my-image   172.30.48.125:5000/test/my-image   latest    3 seconds ago
 ----
 ====
 
+
+[[triggering-updates-on-image-stream-changes]]
+== Triggering Updates On Image Stream Changes
+
+=== Overview
+When an image stream tag is updated to point to a new image, {product-title} can automatically take action to roll the new image out to resources that were using the old image.  This is configured in different ways depending on the type of resource that is referencing the image stream tag.
+
+=== OpenShift Resources
+OpenShift DeploymentConfigs and BuildConfigs can be automatically triggered by changes to ImageStreamTags.  The triggered action can be run using the new value of the image referenced by the updated ImageStreamTag.  For more details on using this capability see the documentation on xref:builds/triggering_builds.adoc#image-change-triggers[BuildConfig triggers] and xref:deployments/basic_deployment_operations.adoc#image-change-trigger[DeploymentConfig triggers].
+
+=== Kubernetes Resources (Alpha)
+Unlike DeploymentConfigs and BuildConfigs, which include as part of their API definition a set of fields for controlling triggers, Kubernetes resources don't have fields for triggering. Instead, {product-title} uses annotations to allow users to request triggering. The annotation is currently alpha and is defined as follows:
+
+====
+----
+Key: image.alpha.openshift.io/triggers
+Value: array of triggers, where each item has the schema:
+[
+ {
+   "from" :{
+     "kind": "ImageStreamTag", // required, the resource to trigger from, must be ImageStreamTag
+     "name": "example:latest", // required, the name of an ImageStreamTag
+     "namespace": "myapp",     // optional, defaults to the namespace of the object
+   },
+   // required, JSON path to change
+   // Note that this field is limited today, and only accepts a very specific set
+   // of inputs (a JSON path expression that precisely matches a container by ID or index).
+   // For pods this would be "spec.containers[?(@.name='web')].image".
+   "fieldPath": "spec.template.spec.containers[?(@.name='web')].image"
+ },
+ ...
+]
+----
+====
+
+
+When {product-title} sees one of the core Kubernetes resources that contains both a pod template (i.e, only CronJobs, Deployments, StatefulSets, DaemonSets, Jobs, ReplicaSets, ReplicationControllers, and Pods) and this annotation, it will attempt to update the object using the image currently associated with the ImageStreamTag referenced by trigger.  The update is performed against the `fieldPath` specified.  
+
+In the following example the trigger will fire when the `example:latest` imagestream tag is updated.  Upon firing, the object's pod template image reference for the `web` container will be updated with a new image value.  If the pod template is part of a Deployment definition that also has a config change trigger, this will further trigger a deployment, effectively rolling out the new image. 
```

Comment:
upstream deployments have config change trigger automatically I believe... any change to a podspec template will trigger a rollout automatically

Reply:
will update.